app/bundles/CampaignBundle/EventListener/CampaignImportExportSubscriber.php

```diff
@@ -0,0 +1,831 @@
+     */
+    private function importCampaigns(EntityImportEvent $event, array $entityData, string $user): void
+    {
+        $stats = [
+            EntityImportEvent::NEW    => ['names' => [], 'ids' => [], 'count' => 0],
+            EntityImportEvent::UPDATE => ['names' => [], 'ids' => [], 'count' => 0],
+        ];
+
+        foreach ($entityData[Campaign::ENTITY_NAME] as $campaignData) {
+            $object = $this->entityManager->getRepository(Campaign::class)->findOneBy(['uuid' => $campaignData['uuid']]);
+            $isNew  = !$object;
+
+            $object ??= new Campaign();
+            $isNew && $object->setDateAdded(new \DateTime());
+            $object->setUuid($campaignData['uuid']);
+            $object->setDateModified(new \DateTime());
+
+            if ($isNew) {
+                $object->setCreatedByUser($user);
+            } else {
+                $object->setModifiedByUser($user);
+            }
+
+            $object->setName(strip_tags($campaignData['name'] ?? '', '<p><b><strong><i><em><u><ul><ol><li><br><span><h1><h2><h3><h4><h5><h6>'));
+            $object->setDescription(strip_tags($campaignData['description'] ?? '', '<p><b><strong><i><em><u><ul><ol><li><br><span><h1><h2><h3><h4><h5><h6>'));
+            $object->setIsPublished(false);
+            $object->setCanvasSettings($campaignData['canvas_settings'] ?? '');
+
+            $this->campaignModel->saveEntity($object);
+
+            $event->addEntityIdMap((int) $campaignData['id'], $object->getId());
+            $status                    = $isNew ? EntityImportEvent::NEW : EntityImportEvent::UPDATE;
+            $stats[$status]['names'][] = $object->getName();
+            $stats[$status]['ids'][]   = $object->getId();
+            ++$stats[$status]['count'];
+
+            $campaignEvent = new CampaignEvent($object, $isNew);
+            $this->dispatcher->dispatch($campaignEvent);
+        }
+
+        foreach ($stats as $status => $info) {
+            if ($info['count'] > 0) {
+                $event->setStatus($status, [Campaign::ENTITY_NAME => $info]);
+            }
+        }
+    }
+
+    public function onDuplicationCheck(EntityImportAnalyzeEvent $event): void
+    {
+        if (Campaign::ENTITY_NAME !== $event->getEntityName() || empty($event->getEntityData())) {
+            return;
+        }
+
+        $summary = [
+            EntityImportEvent::NEW    => ['names' => []],
+            EntityImportEvent::UPDATE => ['names' => [], 'uuids' => []],
+            'errors'                  => [],
+        ];
+        $uuidRegex = '/^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i';
+
+        foreach ($event->getEntityData() as $element) {
+            // UUID format check
+            $uuid = $element['uuid'] ?? '';
+            if (!empty($uuid) && !preg_match($uuidRegex, $uuid)) {
+                $summary['errors'][] = sprintf('Invalid UUID format for %s', $event->getEntityName());
+                break;
```

Comment:
This would be also great to encapsulate in a method and just call it. Please keep the code DRY-er.